app-modules/assistant/src/Filament/Pages/PersonalAssistant.php

```diff
@@ -309,38 +325,61 @@ public function shareChatAction(): Action
             ->action(function (array $arguments, array $data) {
                 $chat = AssistantChat::find($arguments['chat']);
 
-                $users = match ($data['target_type']) {
-                    'team' => collect($data['target_ids'])
-                        ->map(fn ($id) => Team::find($id)->users()->whereKeyNot(auth()->id())->get())
-                        ->flatten()
-                        ->unique(),
-                    'user' => User::whereIn('id', $data['target_ids'])->get(),
+                $users = match ($data['via']) {
+                    AssistantChatShareVia::Email => match ($data['target_type']) {
+                        AssistantChatShareWith::Team => collect($data['target_ids'])
+                            ->map(fn ($id) => Team::find($id)->users)
+                            ->flatten()
+                            ->unique(),
+                        AssistantChatShareWith::User => User::whereIn('id', $data['target_ids'])->get(),
+                    },
+                    AssistantChatShareVia::Internal => match ($data['target_type']) {
+                        AssistantChatShareWith::Team => collect($data['target_ids'])
+                            ->map(fn ($id) => Team::find($id)->users()->whereKeyNot(auth()->id())->get())
+                            ->flatten()
+                            ->unique(),
+                        AssistantChatShareWith::User => User::whereIn('id', $data['target_ids'])->get(),
+                    }
                 };
 
                 $users
```

Comment:
This is potentially a time consuming process if the chat is shared via email to a large team. What are your thoughts on extracting this into an action, and batching (if necessary) a `ShareChat` job? We could do something similar to what happens with Engagement Batches, where a sending (sharing in this case) user is notified when the job starts and finishes, with data covering the expectations (jobs to be processed, jobs processed, jobs failed).